pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -51,6 +53,29 @@ func NewCSIMaxVolumeLimitPredicate(
 	return c.attachableLimitPredicate
 }
 
+func (c *CSIMaxVolumeLimitChecker) getVolumeLimits(nodeInfo *schedulernodeinfo.NodeInfo) map[v1.ResourceName]int64 {
+	// TODO: is it really necessary to read volumes from Node?
```

Comment:
We need this while kubelet is < 1.16 and may not be populating CSINode yet?

Reply:
That's correct, the unit tests are covering this scenario